src/delphi/eval/vis_per_token_model.py

```diff
@@ -0,0 +1,37 @@
+import ipywidgets as widgets
+import matplotlib.pyplot as plt
+import numpy as np
+from ipywidgets import interact
+
+
+def visualize_per_token_category(input):
+    model_names = list(input.keys())
+    categories = list(list(input.values())[0].keys())
+
+    def _f(category):
```

Comment:
more descriptive name, underscore unnecessary (it's already defined inside another function, so you can't import it)

Reply:
type hints